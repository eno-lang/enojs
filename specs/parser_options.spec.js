const eno = require('../eno.js');
const { HtmlReporter, TerminalReporter } = require('../eno.js');

describe('Parser options', () => {
  describe('Non-string input', () => {
    it('throws an error', () => {
      expect(() => eno.parse({ foo: 'bar' })).toThrowErrorMatchingSnapshot();
    });
  });

  describe('Non-object options', () => {
    it('throws an error', () => {
      expect(() => eno.parse('', 'de')).toThrowErrorMatchingSnapshot();
    });
  });

  describe('Unsupported option', () => {
    it('throws an error', () => {
      expect(() => eno.parse('', { wishful: 'thinking' })).toThrowErrorMatchingSnapshot();
    });
  });

  describe('Unsupported locale', () => {
    it('throws an error', () => {
      expect(() => eno.parse('', { locale: 'abcdef' })).toThrowErrorMatchingSnapshot();
    });
  });

  describe('sourceLabel', () => {
    it('is printed in text reports if provided', () => {
      expect(() => eno.parse('boom!', { sourceLabel: '/some/dubious-file.eno' })).toThrowErrorMatchingSnapshot();
    });

    it('is printed in html reports if provided', () => {
      expect(() => eno.parse('boom!', { reporter: HtmlReporter, sourceLabel: '/some/dubious-file.eno' })).toThrowErrorMatchingSnapshot();
    });

    it('is printed in terminal reports if provided', () => {
      expect(() => eno.parse('boom!', { reporter: TerminalReporter, sourceLabel: '/some/dubious-file.eno' })).toThrowErrorMatchingSnapshot();
    });
  });
});
