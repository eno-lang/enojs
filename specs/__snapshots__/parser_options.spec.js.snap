// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser options Non-object options throws an error 1`] = `"Parser options can only be an object, got: de"`;

exports[`Parser options Non-string input throws an error 1`] = `"The parser accepts only strings as input, input was: [object Object]"`;

exports[`Parser options Unsupported locale throws an error 1`] = `"The requested locale 'abcdef' is not available. Translation contributions are greatly appreciated, visit https://github.com/eno-lang/eno-locales if you wish to contribute."`;

exports[`Parser options Unsupported option throws an error 1`] = `"Unknown option 'wishful' supplied. Available ones are: locale, reporter, sourceLabel, zeroIndexing"`;

exports[`Parser options sourceLabel is printed in html reports if provided 1`] = `
"Line 1 does not follow any specified pattern.

<pre class=\\"eno-report\\"><div>/some/dubious-file.eno</div><div class=\\"eno-report-line \\"><div class=\\"eno-report-gutter\\">      Line</div><div class=\\"eno-report-content\\">Content</div></div><div class=\\"eno-report-line eno-report-line-emphasized\\"><div class=\\"eno-report-gutter\\">         1</div><div class=\\"eno-report-content\\">boom!</div></div></pre>"
`;

exports[`Parser options sourceLabel is printed in terminal reports if provided 1`] = `
"Line 1 does not follow any specified pattern.

[1m/some/dubious-file.eno[0m
[1m   Line | Content[0m
[41m >    1 | boom![0m
"
`;

exports[`Parser options sourceLabel is printed in text reports if provided 1`] = `
"Line 1 does not follow any specified pattern.

/some/dubious-file.eno
   Line | Content
 >    1 | boom!
"
`;
